source/Generic/PlayState/PlayState.cs

```diff
@@ -413,6 +411,21 @@ private bool CurrentGameChanged(Game game)
             return false;
         }
 
+        /// <summary>
+        /// Method for obtaining the gameData of the asked game.
+        /// </summary>
+        private PlayStateData GetGameData(Game game)
+        {
+            if (game == null || !playStateData.Any(x => x.Game.Id == game.Id))
+            {
+                return null;
+            }
+            else
+            {
+                return playStateData.Find(x => x.Game.Id == game.Id);
+            }
```

Comment:
I think this could be simplified by just using:

```csharp
return playStateData.FirstOrDefault(x => x.Game.Id == game.Id)
```

Currently it's enumeration twice, one for the `Any` and then in the `Find`. With what I propose it would just search once and if it doesn't find anything, it will return `null` anyway